service/types.go

```diff
@@ -286,3 +288,8 @@ func (m *accountMetadata) UnmarshalJSON(data []byte) error {
 	m.Nonce = nonce
 	return nil
 }
+
+// has0xPrefix validates str begins with '0x' or '0X'.
+func has0xPrefix(str string) bool {
```

Comment:
Nit: is there not a helper already in eth to do this? if not, you should use `strings.HasPrefix`

Reply:
This point still stands.